VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -81,9 +81,8 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
                 discussionThread.getIdentifier(),
                 discussionThread.getType() == DiscussionThread.ThreadType.QUESTION);
 
-        // Using application context to prevent activity leak since adapter is retained across config changes
         courseDiscussionResponsesAdapter = new CourseDiscussionResponsesAdapter(
-                getActivity().getApplicationContext(), this, discussionThread);
+                getActivity(), this, discussionThread);
```

Comment:
For the short term, we should ensure that the Adapters don't hold a persistent reference to an old instance of the Activity that has been reinstantiated since. This could be done by just having a setter for the `TaskProcessCallback` in the adapter, and having the Fragment set and remove it at the appropriate points in it's Activity lifecycle callbacks, or more simply by just passing a reference to the Fragment to the Adapter instead of a Context, and having it get the Context from the Fragment at each point it needs to get one.

In the long term, as we implement the MVP framework across the app, we will decouple the data from the views, and the Fragments will no longer need to be retained. This would simplify the overall logic and reduce such hidden catches by quite a bit.